src/rgw/rgw_bucket.cc

```diff
@@ -1550,32 +1550,28 @@ int RGWBucketAdminOp::limit_check(RGWRados *store,
 	  continue;
 
 	for (const auto& s : stats) {
-	    num_objects += s.second.num_objects;
+	  num_objects += s.second.num_objects;
 	}
 
 	num_shards = info.num_shards;
 	uint64_t objs_per_shard =
 	  (num_shards) ? num_objects/num_shards : num_objects;
 	{
-	  bool warn = false;
+	  bool warn;
 	  stringstream ss;
-	  if (objs_per_shard > safe_max_objs_per_shard) {
-	    double over =
-	      100 - (safe_max_objs_per_shard/objs_per_shard * 100);
-	      ss << boost::format("OVER %4f%%") % over;
-	      warn = true;
+	  uint64_t fill_pct = objs_per_shard * 100 / safe_max_objs_per_shard;
+	  if (fill_pct > 100) {
+	    ss << "OVER " << fill_pct << "%";
+	    warn = true;
+	  } else if (fill_pct >= shard_warn_pct) {
+	    ss << "WARN " << fill_pct << "%";
```

Comment:
is removing formatting really preferable?

Reply:
before: `xy.00%`, after: `xy%`

Sure, i could change it to `xy.za%`, but I don't think that's helpful.

I use this command quite frequently and I really don't care if it's 110% or 110.35%, it just makes it harder to read.